src/App.jsx

```diff
@@ -14,45 +15,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD = {
+  SORT_ALPHABETICALLY: 'Sort alphabetically',
+  SORT_BY_LENGTH: 'Sort by length',
+  REVERSE: 'Reverse',
+};
+
+export const App = () => {
+  const [arrayValues, setArrayValues] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortByAlphabet = () => {
+    const alphabetSort = [...arrayValues].sort();
+
+    if (isReversed) {
+      alphabetSort.reverse();
+    }
+
+    setArrayValues(alphabetSort);
+    setSortField(SORT_FIELD.SORT_ALPHABETICALLY);
+  };
+
+  const sortByLength = () => {
+    const lenghtSort = [...goodsFromServer].sort(
+      (value1, value2) => value1.length - value2.length,
+    );
+
+    if (isReversed) {
+      lenghtSort.reverse();
+    }
+
+    setArrayValues(lenghtSort);
+    setSortField(SORT_FIELD.SORT_BY_LENGTH);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    setArrayValues(goodsFromServer);
+    setSortField('');
+
+    if (isReversed) {
+      setIsReversed(false);
+    }
+  };
+
+  const reverse = () => {
+    setArrayValues([...arrayValues].reverse());
+    setIsReversed(current => !current);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={sortByAlphabet}
+          type="button"
+          className={sortField === SORT_FIELD.SORT_ALPHABETICALLY ? (
+            'button is-info'
+          ) : (
+            'button is-info is-light'
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={sortByLength}
+          type="button"
+          className={sortField === SORT_FIELD.SORT_BY_LENGTH ? (
+            'button is-success'
+          ) : (
+            'button is-success is-light'
```

Comment:
no need to overwrite all class names, only `is-light`
you can use [classnames](https://www.npmjs.com/package/classnames) library